File: frontend/src/types/document.ts
export interface OCRField {
  value: string;
  confidence: number;
  bounding_box?: [number, number, number, number];
}

export interface OCRResult {
  document_type: string;
  fields: Record<string, OCRField>;
  raw_ocr: Array<{
    text: string;
    confidence: number;
    bounding_box: [number, number, number, number];
  }>;
  overall_confidence: number;
}

export interface MRZFieldComparison {
  ocr?: string;
  mrz?: string;
  consistent: boolean;
}

export interface MRZResult {
  detected: boolean;
  raw_mrz: string[];
  document_code?: string;
  issuing_country?: string;
  surname?: string;
  given_names?: string;
  passport_number?: string;
  nationality?: string;
  date_of_birth?: string;
  sex?: string;
  expiry_date?: string;
  check_digits_valid: boolean;
  field_comparisons: Record<string, MRZFieldComparison>;
}

File: backend/app/models/document.py
from pydantic import BaseModel, Field
from typing import Dict, Any, Optional, List

class OCRField(BaseModel):
    value: str
    confidence: float = 1.0
    bounding_box: Optional[List[int]] = None # [x, y, w, h]

class OCRResult(BaseModel):
    document_type: str
    fields: Dict[str, OCRField] = {}
    raw_ocr: List[Dict[str, Any]] = []
    overall_confidence: float = 0.95

class MRZFieldComparison(BaseModel):
    ocr: Optional[str] = None
    mrz: Optional[str] = None
    consistent: bool = True

class MRZResult(BaseModel):
    detected: bool = False
    raw_mrz: List[str] = []
    document_code: Optional[str] = None
    issuing_country: Optional[str] = None
    surname: Optional[str] = None
    given_names: Optional[str] = None
    passport_number: Optional[str] = None
    nationality: Optional[str] = None
    date_of_birth: Optional[str] = None
    sex: Optional[str] = None
    expiry_date: Optional[str] = None
    check_digits_valid: bool = True
    field_comparisons: Dict[str, MRZFieldComparison] = {}
